### PR TITLE
Celx LUA value to Celestia parser Value convertion helper.

### PR DESCRIPTION
**Pull request by pirogronian**

### CODE REVIEW

File: src/celestia/celx.cpp [hunk 1]
```diff
@@ -1798,6 +1798,61 @@ bool CelxLua::isType(int index, int type) const
...
+                {
+                    delete hash;
+                    hash = nullptr;
+                }
+                if (array != nullptr)
+                {
+                    array->push_back(getValue(-1));
+                }
+            }
+            if (isString(-2))
```
Comment (375gnu) [inline src/celestia/celx.cpp]: `else`

File: src/celestia/celx.cpp [hunk 2]
```diff
@@ -1798,6 +1798,61 @@ bool CelxLua::isType(int index, int type) const
...
+                    array = nullptr;
+                }
+                if (hash != nullptr)
+                {
+                    hash->addValue(getString(-2), *getValue(-1));
+                }
+            }
+            pop(1);
+            if (array == nullptr &&
+                hash == nullptr)
```
Comment (375gnu) [inline src/celestia/celx.cpp]: don't break line

File: src/celestia/celx.cpp [hunk 3]
```diff
@@ -1798,6 +1798,61 @@ bool CelxLua::isType(int index, int type) const
...
+                break;
+        }
+        pop(1);
+        if (hash != nullptr)
+            v = new Value(hash);
+        if (array != nullptr)
```
Comment (375gnu) [inline src/celestia/celx.cpp]: `else`

**Comment by pirogronian** [comment]
As LUA don't separate arrays and hash tables, I check keys integrity. If table has only integer keys, then become an Array. Ih has only string keys, become an Hash. Otherwise table is invalid and function returns null pointer.

**Pull request merged**